core/domain/exp_domain_test.py

```diff
@@ -507,6 +507,20 @@ def test_checkpoint_count_with_count_outside_range_is_invalid(self):
         self.exploration.validate()
 
     def test_bypassable_state_with_card_is_checkpoint_true_is_invalid(self):
+        # Exploration to test a checkpoint state which has no outcome.
```

Comment:
Indicate in your diagrams which states are checkpoints, and explain the legend used. (Maybe add a * to the end of checkpoint states.)

Reply:
Done